src/routes/package-create-route/package-create-route.test.js

```diff
@@ -0,0 +1,162 @@
+
+describe('Given PackageCreateRoute', () => {
+  beforeEach(() => {
+    mockGetAccessTypes.mockClear();
+    mockCreatePackage.mockClear();
+    mockRemoveCreateRequests.mockClear();
+    mockHistory.replace.mockClear();
+  });
+
+  afterEach(cleanup);
+  it('should render PackageCreateRoute', async () => {
+    const { getByText } = renderPackageCreateRoute();
+
+    expect(getByText('Page content')).toBeDefined();
+  });
+
+  it('should handle getAccessTypes', async () => {
+    await renderPackageCreateRoute({
+      props: { getAccessTypes: mockGetAccessTypes },
+    });
+
+    expect(mockGetAccessTypes).toHaveBeenCalled();
+  });
+
+  describe('when request is not resolved', () => {
+    it('should not redirect to new package record', () => {
+      const { rerender } = renderPackageCreateRoute();
+
+      rerender(getPackageCreateRoute({
+        createRequest: {
+          ...createRequest,
+          isResolved: false,
+        },
+      }));
+
+      expect(mockHistory.replace).toHaveBeenCalledTimes(0);
+    });
+  });
+
+  describe('when request is resolved', () => {
+    it('should redirect to new package record', () => {
+      const { rerender } = renderPackageCreateRoute();
+
+      rerender(getPackageCreateRoute({
+        createRequest: {
+          ...createRequest,
+          isResolved: true,
+        },
+      }));
+
+      expect(mockHistory.replace).toHaveBeenCalled();
+    });
+  });
+
+  describe('when click on close icon and form is not pristine', () => {
+    it('should handle history.goBack', () => {
+      const { getByRole } = renderPackageCreateRoute({
+        location: {
+          ...location,
+          state: { eholdings: true },
+        },
+      });
+
+      const packageNameInput = getByRole('textbox', { name: 'ui-eholdings.label.name' });
+
+      fireEvent.change(packageNameInput, { target: { value: 'New package name' } });
+      fireEvent.blur(packageNameInput);
+
+      fireEvent.click(getByRole('button', { name: 'ui-eholdings.label.icon.closeX' }));
+
+      expect(mockHistory.goBack).toHaveBeenCalled();
+    });
+  });
+
+  describe('when submit form with some values and click save', () => {
+    it('should handle mockCreatePackage action', async () => {
+      const { getByRole, getByText, getByTestId } = renderPackageCreateRoute();
+
+      const packageNameInput = getByRole('textbox', { name: 'ui-eholdings.label.name' });
+
+      fireEvent.change(packageNameInput, { target: { value: 'New package name' } });
+      fireEvent.blur(packageNameInput);
+
+      const addCoverageSettingsButton = getByText('ui-eholdings.package.coverage.addDateRange');
+      fireEvent.click(addCoverageSettingsButton);
+
+      fireEvent.change(getByTestId('begin-coverage-0'), { target: { value: '01/01/2022' } });
+      fireEvent.change(getByTestId('end-coverage-0'), { target: { value: '10/01/2022' } });
+      getByTestId('begin-coverage-0').blur();
+      getByTestId('end-coverage-0').blur();
```

Comment:
Maybe it's better to use `fireEvent.blur()` here, since you use it in the test above, to keep the code consistent?
```suggestion
      fireEvent.blur(getByTestId('begin-coverage-0'));
      fireEvent.blur(getByTestId('end-coverage-0'));
```